Domain/Entities/CoSignatureDocument.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Domain.Entities
+{
+    public class CoSignatureDocument
```

Comment:
Please create an abstract Document class from the current Document.cs and inherit from it. Also an OmItemDocument class should be created inherinting from the abstract Document